api/transport/header.go

```diff
@@ -102,6 +109,16 @@ func (h Headers) Items() map[string]string {
 	return h.items
 }
 
+// ExactCaseItems returns the non-canonicalized version of the underlying map
+// for this Headers object. The returned map MUST NOT be changed.
+// Doing so will result in undefined behavior.
+func (h Headers) ExactCaseItems() map[string]string {
+	if h.originalItems == nil {
+		return emptyMap
```

Comment:
unrelated question for @willhug and @abhinav, why do we return a non-nil map here? `nil` is a valid read-only map, and if the user tries to modify it (which the header explicitly warns them against), it will crash.

Reply:
It's possible that when this was originally done, we didn't realize `nil` was a valid map. I'm not sure of the exact reasons for this. This should be safe to delete. (@AllenLuUber Don't worry about it. We'll take care of it.)

Reply:
Fixed in https://github.com/yarpc/yarpc-go/pull/1433